### ISSUE TITLE
Allow variable tab number on jsx-indent-props

### ISSUE THREAD

**Issue by matthewheck** [yannickcr/eslint-plugin-react#401]
Just setting up some eslint rules for React today and we realized that our style didn't quite work.

We are using `prop-aligned` for `jsx-closing-bracket-location` and `tab` for `jsx-indent-props`. This leads to a somewhat hard to parse line of code:

```
<Hello
    firstName="Matthew"
    lastName="Heck">
    <Status />
</Hello>
```

We tried this, but putting the nested component on the same line as the props just doesn't work for us (and we also don't like the alternative closing bracket locations).

I was hoping to propose that this be allowed:

```
<Hello
        firstName="Matthew"
        lastName="Heck">
    <Status />
</Hello>
```

I believe the solution would be fairly trivial (allowing a number of tabs) and can be implemented in a way that ensures backwards compatibility. I am happy to implement this change myself, but wanted to get feedback from you all first.

**Comment by yannickcr** [comment]
Does something like #398 can suit your need ?

**Comment by ljharb** [comment]
I think that if you're choosing tabs (which imo is the proper choice), you've already chosen to not use line-aligned, tag-aligned, or prop-aligned anything - because the only way anything should be "aligned" is by semantic indentation level.

It's only with space-based indentation that it would be feasible to have arbitrary indentation amounts (in a granularity smaller than a whole number of indentation levels).
